src/Hazelcast.Net/HazelcastClient.Objects.cs

```diff
@@ -25,6 +26,12 @@ namespace Hazelcast
     internal partial class HazelcastClient // Distributed Objects
     {
         private readonly ISequence<long> _lockReferenceIdSequence = new Int64Sequence();
+        private ICPSubsystem _cpSubsystem;
+
+        /// <summary>
+        /// Gets the CP subsystem.
+        /// </summary>
+        public ICPSubsystem CPSubsystem => _cpSubsystem ??= new CPSubsystem(Cluster);
```

Comment:
I am not familiar with C#, but is `??=` safe to use from multiple tasks? I mean, can I end up with multiple `CPSubsystem` objects from different tasks? Maybe this is not a problem as of now, since we don't have any state on CP subsystem, but once you implement FencedLock, it might be problematic

Reply:
`??=` is supposed to be safe, yes - and that code is internal and can change later on - one question I had though is... do we agree just creating the `ICPSubSystem` instance is *always* a local operation that does not involve the server? I am using a property here so it has to be synchronous. My understanding of the Java code is that it is OK, and it's only when you create a FencedLock for instance, that you talk to the server. Can you confirm this?

Reply:
It should be OK, creating CP Subsystem is always local, operations on it might talk with the server as you said